### DISCUSSION TITLE
"gridy" and "jdenticon" avatar styles are now deprecated

### DISCUSSION THREAD

**Discussion by FlorianKoerner** [dicebear/dicebear#187] [Announcements]
The avatar styles `gridy` and `jdenticon` are now deprecated. They are based on the great work of [darosh](https://github.com/darosh/gridy-avatars) and [dmester](https://github.com/dmester/jdenticon). I want to keep [version 5.0](https://github.com/dicebear/dicebear/pull/153) completely dependency free. At the same time there is no reason to copy the libraries, because you can just use their libraries directly. I want to explain how to switch.

# gridy

```sh
# Install the package with npm
npm install gridy-avatars prando
```

```js
// In your code
import { outer } from 'gridy-avatars/dist/avatars';
import Prando from 'prando';

const seed = "your-custom-seed";
const prng = new Prando(seed);

const body = prng.nextInt(0, 7);
const bodyColor = prng.nextInt(0, 7);

const eyes = prng.nextInt(0, 7);
const eyesColor = prng.nextInt(0, 7);

const mouth = prng.nextInt(0, 7);
const mouthColor = prng.nextInt(0, 7);

const avatar = outer(`${body}${bodyColor}${eyes}${eyesColor}${mouth}${mouthColor}`);
```

Documentation: https://github.com/darosh/gridy-avatars#usage

# jdenticon

```sh
# Install the package with npm
npm install jdenticon
```

```js
// In your code
import { toSvg } from "jdenticon";

const seed = "your-custom-seed";
const width = 50;

const avatar = toSvg(seed, 50);
```

Documentation: https://jdenticon.com/get-started/typescript.html

**Comment by Alfagun74** [comment]
very sad. Will the API still have these? @FlorianKoerner

**Reply by FlorianKoerner** [reply]
Yes.
